### ISSUE TITLE
getEntityNames fails if no roles come in on request

### ISSUE THREAD

**Issue by jewzaam** [lightblue-platform/lightblue-rest#67]
If metadata roles are configured but the request doesn't include any roles it fails with a `NullPointerException` trying to loop over the null `roles` field.  This is true for version `1.2.0`.  Looking at the latest `1.3.0-SNAPSHOT` code it has been changed but looks like it would still be a problem, the field `roles` is never checked.

https://github.com/lightblue-platform/lightblue-rest/blob/6bcb7414ff0e3418eef54fd7df0d3c477f35f4c3/metadata/src/main/java/com/redhat/lightblue/rest/metadata/AbstractMetadataResource.java#L242-L255

**Comment by vritant** [comment]
@jewzaam  does the NPE occur within the isUserInRole method invocation ?  asking cause if the If the roleMap element is defined, it is expected to have all the roles mapped

**Comment by jewzaam** [comment]
@vritant this commit reproduces the issue if I remember correctly: https://github.com/jewzaam/openshift-lightblue-cart/commit/9d8c31c75cb24f11042dd5384629391a901996e5

**Comment by vritant** [comment]
just a status update :
was able to re-produce this:

curl http://vrlblue-vritant.rhcloud.com/rest/metadata

 will investigate further.

**Comment by vritant** [comment]
@jewzaam  This is no longer an issue for v: 1.3.0-SNAPSHOT 

deployed the latest jar to my open shift gear, and found that the issue no longer occurs. The call :
curl http://vrlblue-vritant.rhcloud.com/rest/metadata
now fails with the message :
  java.lang.SecurityException: Unauthorized Request. One of the following roles is required: [<ROLE NAME>]

**Comment by vritant** [comment]
( the error also lists the role name required )

**Comment by jewzaam** [comment]
@vritant you are testing with something that doesn't deploy `roleMap` attribute in `lightblue-metadata.json`.  I just reproduced this with:

```
rhc app create http://cartreflect-claytondev.rhcloud.com/github/jewzaam/openshift-lightblue-cart?commit=9d8c31c75cb24f11042dd5384629391a901996e5 -a testnpe2
curl http://testnpe2-nmalikit.rhcloud.com/rest/metadata/
```

And got
```
<html><head><title>JBoss Web/7.4.9.Final-redhat-1 - JBWEB000064: Error report</title><style><!--H1 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:22px;} H2 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:16px;} H3 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:14px;} BODY {font-family:Tahoma,Arial,sans-serif;color:black;background-color:white;} B {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;} P {font-family:Tahoma,Arial,sans-serif;background:white;color:black;font-size:12px;}A {color : black;}A.name {color : black;}HR {color : #525D76;}--></style> </head><body><h1>JBWEB000065: HTTP Status 500 - java.lang.NullPointerException</h1><HR size="1" noshade="noshade"><p><b>JBWEB000309: type</b> JBWEB000066: Exception report</p><p><b>JBWEB000068: message</b> <u>java.lang.NullPointerException</u></p><p><b>JBWEB000069: description</b> <u>JBWEB000145: The server encountered an internal error that prevented it from fulfilling this request.</u></p><p><b>JBWEB000070: exception</b> <pre>org.jboss.resteasy.spi.UnhandledException: java.lang.NullPointerException
        org.jboss.resteasy.core.SynchronousDispatcher.handleApplicationException(SynchronousDispatcher.java:365)
        org.jboss.resteasy.core.SynchronousDispatcher.handleException(SynchronousDispatcher.java:233)
        org.jboss.resteasy.core.SynchronousDispatcher.handleInvokerException(SynchronousDispatcher.java:209)
        org.jboss.resteasy.core.SynchronousDispatcher.getResponse(SynchronousDispatcher.java:557)
        org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:524)
        org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:126)
        org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:208)
        org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:55)
        org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:50)
        javax.servlet.http.HttpServlet.service(HttpServlet.java:847)
</pre></p><p><b>JBWEB000071: root cause</b> <pre>java.lang.NullPointerException
        com.redhat.lightblue.rest.metadata.AbstractMetadataResource.getEntityNames(AbstractMetadataResource.java:135)
        sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        java.lang.reflect.Method.invoke(Method.java:606)
        org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:167)
        org.jboss.resteasy.core.ResourceMethod.invokeOnTarget(ResourceMethod.java:269)
        org.jboss.resteasy.core.ResourceMethod.invoke(ResourceMethod.java:227)
        org.jboss.resteasy.core.ResourceMethod.invoke(ResourceMethod.java:216)
        org.jboss.resteasy.core.SynchronousDispatcher.getResponse(SynchronousDispatcher.java:542)
        org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:524)
        org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:126)
        org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:208)
        org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:55)
        org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:50)
        javax.servlet.http.HttpServlet.service(HttpServlet.java:847)
</pre></p><p><b>JBWEB000072: note</b> <u>JBWEB000073: The full stack trace of the root cause is available in the JBoss Web/7.4.9.Final-redhat-1 logs.</u></p><HR size="1" noshade="noshade"><h3>JBoss Web/7.4.9.Final-redhat-1</h3></body></html>
```

**Comment by vritant** [comment]
@jewzaam the I think commit 9d8c31c75cb24f11042dd5384629391a901996e5 uses version 1.2 . Please let me know if that is not so.
I deployed lightblue-rest-metadata-1.3.0-SNAPSHOT.war and lightblue-rest-crud-1.3.0-SNAPSHOT.war to the open shift gear, and updated lightblue-metadata.json with the expected roles.

please run curl http://vrlblue-vritant.rhcloud.com/rest/metadata 
the error message we get is :
 java.lang.SecurityException: Unauthorized Request. One of the following roles is required: [anyone2]

which means it is reading the roles configuration ( anyone2 is the role name ) , and failing when the request does not have the role configured, no NPE.

**Comment by jewzaam** [comment]
Found that the latest artifacts were not being published.  And then that when they got published we had changed the names of everything.  Working to resolve that before this can issue can be confirmed.  Likely won't be done until late tonight.

**Comment by jewzaam** [comment]
@vritant I have fixed the issues with the cart.  I'll document it on https://github.com/lightblue-platform/openshift-lightblue-cart/pull/9 but basically  it now pulls down the correct versions of stuff.

I have verified as much as I can on rhcloud right now.  Did a complete verification on our open shift enterprise deployment but for some reason rhc is not happy with rhcloud right now.

Glad I started digging, it uncovered some issues with the cart PR!

Thanks for looking, @vritant.
